### DISCUSSION TITLE
How to change html output? "#" is converted to <h1>, i want to change this to <h2>.

### DISCUSSION THREAD

**Discussion by V1nc3VD** [spatie/laravel-markdown#21] [Q&A] [answered]
Hi, i think i need to change the rendering process, described here: https://spatie.be/docs/laravel-markdown/v1/advanced-usage/customizing-the-rendering-process but this is not very clear for me who just started with laravel recently.

How do i do this? I understand i need to make a new class that overrides the old one. Do i place that new class in providers? Or in vendor or somewhere else?

I would like to change the html output of headings.

Thanks a lot.

**Comment by lnfel** [comment, marked as answer]
Hi, in Markdown `#` have equivalent value in html which is `h1`, `##` is `h2`, `###` is `h3` and so on. Keep in mind that html only had up to `h6`.
To render `h2` try using `##`. For example:
```
## Your h2 heading
```

**Comment by V1nc3VD** [comment]
> Hi, in Markdown `#` have equivalent value in html which is `h1`, `##` is `h2`, `###` is `h3` and so on. Keep in mind that html only had up to `h6`. To render `h2` try using `##`. For example:
> 
> ```
> ## Your h2 heading
> ```

Hi thanks for the answer. Users can input markdown code. What i want is to prevent users from using h1 tags. Is there not any easy way to change the output of the markdown? It either needs to be not converted to html or converted to something else than h1.

**Comment by riasvdv** [comment]
> Hi thanks for the answer. Users can input markdown code. What i want is to prevent users from using h1 tags.

Preventing this is outside of the scope of this package. You can replace the content after it is generated, or add validation before you let users submit.
